File: abc324/abc324_a.py
# ------------------------------------------------------------
#
#
# 	   Welcome to my program!
# 	   DON'T HACK PLEASE!!!!!!!!
#
# 	　　∧＿∧        AtCoder / Codeforces
# 	　 ( 　･ω･)
# 	＿(__つ/￣￣￣ /  MIT License
# 	　　＼/　　　　 /  Python 3.10.12
# 	　　　　￣￣￣￣￣
# 		   Let's write Code!
#
#
# ------------------------------------------------------------

import numpy as np
import collections
import math


def main():
    N = int(input())
    A = list(map(int, input().split()))

    a = A[0]
    flag = True
    for i in range(N):
        if a != A[i]:
            flag = False
            break

    print("Yes" if flag else "No")


if __name__ == "__main__":
    main()

File: abc324/abc324_a.ts
/*------------------------------------------------------------



	   Welcome to my program!

	  ∧＿∧        AtCoder / Codeforces
	 ( 　･ω･)
	＿(__つ/￣￣￣ /  MIT License
	  ＼/　　　　 /   TypeScript 5.2.2 + Node.js v18.18.0
		￣￣￣￣￣
		   Let's write Code!


------------------------------------------------------------*/



//#region Lib

import { readFileSync } from 'fs';
import * as mathjs from 'mathjs';
import * as std from 'tstl';


const dx: Array<number> = [0, 0, 1, -1];
const dy: Array<number> = [1, -1, 0, 0];

// #endregion

let N: number, M: number;
let H: number, W: number;
let S: string = "";

// Functions

let A: number[];

function inputs(inp: string[][]): void {
	N = Number(inp[0][0]);
	A = inp[1].map(Number);
}

function main(): void {
	let a: number = A[0];
	let flag: boolean = true;
	for (let i = 0; i < N; ++i) {
		if (a != A[i]) {
			flag = false;
			break;
		}
	}

	console.log(flag ? "Yes" : "No");
}

// #region 入力

inputs(readFileSync("/dev/stdin", "utf-8").split("\n").map((line: string): string[] => {
	return line.split(" ");
}));

main();

// #endregion
